### PR TITLE
Document using versus import

### PR DESCRIPTION
**Pull request by rikhuijzer**
@lazarusA, in response to using Plots and Makie at the same time. I think that the most future proof solution is to use

```
@reexport begin

using Plots:
    plot,
    [...]
using Makie:
    density,
    [...]

end # @reexport
```

The rationale is added to the notation section and can also be read in the [JuMP Style Guide](https://jump.dev/JuMP.jl/stable/developers/style/#using-vs.-import).

As a proof-of-concept, I've implemented it for `DataFrames.jl` and `XLSX.jl` already.

This pattern is also used at other places. For example, in [`JETTest.jl`](https://github.com/aviatesk/JETTest.jl/blob/master/src/JETTest.jl).

### CODE REVIEW

**Comment by storopoli** [comment]
We might need to do this for all sections.
Also, why is the build failing?

File: src/JDS.jl [hunk 1]
```diff
@@ -4,28 +4,52 @@ import XLSX
...
+using FileIO
+using GLMakie
+using GeometryBasics
+using InteractiveUtils
+using LaTeXStrings
+using LinearAlgebra
+using Makie
+using Random
+using Statistics
+using StatsBase
```
Comment (storopoli) [inline src/JDS.jl:47]:
```suggestion
using StatsBase:
    mad,
    mode
```

**Review by storopoli — APPROVED** [review]

File: pandoc/references.bib [hunk 2]
```diff
@@ -70,7 +70,7 @@ @misc{johnmyleswhiteJuliaDataDataFramesJl2020
 @misc{jump2021using,
   title = {JuMP Style Guide},
   year = 2021,
-  howpublished = {\url{https://jump.dev/JuMP.jl/stable/developers/style/#using-vs.-import}},
```
Comment (storopoli) [inline pandoc/references.bib]: Hardcoding versions?

Comment (rikhuijzer) [reply pandoc/references.bib]: I just did that in 7acc2f9

Comment (rikhuijzer) [reply pandoc/references.bib]: For the docs it makes sense, right?

Comment (storopoli) [reply pandoc/references.bib]: What if in year from now JUMP is on v1? Do we need to keep track of JuMP versions?

Comment (rikhuijzer) [reply pandoc/references.bib]: It's only a ref to the information stored in that style guide. That information is still valid when JuMP is at a newer version.

Comment (storopoli) [reply pandoc/references.bib]: OK

**Comment by storopoli** [comment]
“Prefer to load symbols” sounds like `:A`. I think we need to explain what “symbols” are in this context.

**Review by storopoli — APPROVED** [review]

**Review by lazarusA — APPROVED** [review]

**Pull request merged**